hudi-sync/hudi-dla-sync/src/main/java/org/apache/hudi/dla/DLASyncTool.java

```diff
@@ -0,0 +1,211 @@
+  public static final String SUFFIX_SNAPSHOT_TABLE = "_rt";
+  public static final String SUFFIX_READ_OPTIMIZED_TABLE = "_ro";
+
+  private final DLASyncConfig cfg;
+  private final HoodieDLAClient hoodieDLAClient;
+  private final String snapshotTableName;
+  private final Option<String> roTableTableName;
+
+  public DLASyncTool(Properties properties, FileSystem fs) {
+    super(properties, fs);
+    this.hoodieDLAClient = new HoodieDLAClient(Utils.propertiesToConfig(properties), fs);
+    this.cfg = Utils.propertiesToConfig(properties);
+    switch (hoodieDLAClient.getTableType()) {
+      case COPY_ON_WRITE:
+        this.snapshotTableName = cfg.tableName;
+        this.roTableTableName = Option.empty();
+        break;
+      case MERGE_ON_READ:
+        this.snapshotTableName = cfg.tableName + SUFFIX_SNAPSHOT_TABLE;
+        this.roTableTableName = cfg.skipROSuffix ? Option.of(cfg.tableName) :
+            Option.of(cfg.tableName + SUFFIX_READ_OPTIMIZED_TABLE);
+        break;
+      default:
+        LOG.error("Unknown table type " + hoodieDLAClient.getTableType());
+        throw new InvalidTableException(hoodieDLAClient.getBasePath());
+    }
+  }
+
+  @Override
+  public void syncHoodieTable() {
+    try {
+      switch (hoodieDLAClient.getTableType()) {
+        case COPY_ON_WRITE:
+          syncHoodieTable(snapshotTableName, false);
+          break;
+        case MERGE_ON_READ:
+          // sync a RO table for MOR
+          syncHoodieTable(roTableTableName.get(), false);
+          // sync a RT table for MOR
+          syncHoodieTable(snapshotTableName, true);
+          break;
+        default:
+          LOG.error("Unknown table type " + hoodieDLAClient.getTableType());
+          throw new InvalidTableException(hoodieDLAClient.getBasePath());
+      }
+    } catch (RuntimeException re) {
+      LOG.error("Got runtime exception when dla syncing", re);
+    } finally {
+      hoodieDLAClient.close();
+    }
+  }
+
+  private void syncHoodieTable(String tableName, boolean useRealtimeInputFormat) {
+    LOG.info("Trying to sync hoodie table " + tableName + " with base path " + hoodieDLAClient.getBasePath()
+        + " of type " + hoodieDLAClient.getTableType());
+    // Check if the necessary table exists
+    boolean tableExists = hoodieDLAClient.doesTableExist(tableName);
+    // Get the parquet schema for this table looking at the latest commit
+    MessageType schema = hoodieDLAClient.getDataSchema();
+    // Sync schema if needed
+    syncSchema(tableName, tableExists, useRealtimeInputFormat, schema);
+
+    LOG.info("Schema sync complete. Syncing partitions for " + tableName);
+    // Get the last time we successfully synced partitions
+    Option<String> lastCommitTimeSynced = Option.empty();
+    /*if (tableExists) {
+      lastCommitTimeSynced = hoodieDLAClient.getLastCommitTimeSynced(tableName);
+    }*/
+    LOG.info("Last commit time synced was found to be " + lastCommitTimeSynced.orElse("null"));
```

Comment:
since dla meta do not support alter table properties, it would be simpler here

Reply:
yes